server/api/utils/clients/iguazio.py

```diff
@@ -72,6 +74,87 @@ def all():
         ]
 
 
+class JobCache:
+    """
+    Cache for create/delete project jobs.
+    This cache is used to avoid consecutive create/delete jobs for the same project,
+    while the jobs are still in progress.
+    """
+
+    def __init__(self, ttl: str):
+        self._delete_jobs = {}
+        self._create_jobs = {}
+        self._delete_locks = {}
+        self._create_locks = {}
+
+        self._ttl = humanfriendly.parse_timespan(ttl)
+
+    def get_delete_lock(self, project: str):
+        if project not in self._delete_locks:
+            self._delete_locks[project] = threading.Lock()
```

Comment:
what if you get a context switch between those lines?

Reply:
locked 🔒